lib/gollum-lib/page.rb

```diff
@@ -466,6 +506,28 @@ def page_match(name, path)
       false
     end
 
+    # Private: get the node where the header is located
+    #
+    # Extracts the header node from doc.
+    #
+    # TODO: this case is a bad encapsulation, the markup
```

Comment:
I say that the formaters should own this knowledge, so this should simply forward the call to the correct formater. Another option would be to have a special class to handle the title concern, I would prefer this, as the concerns seem to be loosely distribute along different classes. A normal thing to happen when projects have many devs and evolve more or less chaotic.

On my opinion the formater have a loose contract with the rest of the wiki in terms of html generation. I understand the "why" this is like that, so that the assimilation of new formats is simpler. But I think this is the root cause of the problem here. If the rest of the system could have more certain that this would simply be the the first `h1` or that it would be marked as class/id `title` for instance, this code would be simpler.